### DISCUSSION TITLE
Active Chain Issues

### DISCUSSION THREAD

**Discussion by igorsyl** [stacks-network/stacks-core#3644] [General]
[Active Chain Issues Google Doc](https://docs.google.com/document/d/1QflB3K5OSjVBJ7DGgFRJwUQT691A3WnFK0qAmd7wxJM/edit#)

The entity created 10 siblings to overrule the smaller miners with their fork.

from @peters.btc (Discord Conversation Link) GitHub: @Ston3cutt3r
https://discord.com/channels/621759717756370964/730814571517968436/1091340722143252510
[Supporting Documentation Google Sheet](https://docs.google.com/spreadsheets/d/1lc1o8zZDLiJ2xp9aHR9MeAJ0MIvXFalWtkVHefZKRbs/edit#gid=1829578615)

![image](https://user-images.githubusercontent.com/1248500/229124484-fea22916-7074-473f-81c2-ed797d6fbf29.png)

**Comment by kantai** [comment]
Useful sqlite query for dumping block commit information:

```sql
SELECT
   block_headers.burn_header_height as burn_height,
   b.index_block_hash as index_block_hash,
   payments.address as miner,
   block_headers.parent_block_id as parent_ibh,
   block_headers.block_height as stacks_height,
   block_headers.burn_header_timestamp as burn_time,
   b.arrival_time as arrival_time,
   b.processed_time as processed_time
FROM staging_blocks as b 
INNER JOIN block_headers ON block_headers.index_block_hash = b.index_block_hash
INNER JOIN payments ON payments.index_block_hash = b.index_block_hash
WHERE payments.miner = 1 GROUP BY block_headers.burn_header_height
ORDER BY burn_height DESC;
```

This can be executed against the `mainnet/chainstate/vm/index.sqlite` database to get a dump of information about the block selected by each burn block. E.g.,

```bash
cat query_info.sql | sqlite3 -csv mainnet/chainstate/vm/index.sqlite > ~/Documents/block_data_dump_1.csv
```

**Comment by igorsyl** [comment]
From @wileyj, to dump peer node IPs:

Take this address: SP3HDY11ZDHYTMP6A850RW4EM7YQCPFP9WMEXFZHK, decode it to a public key, then simply grep that key in a neighbors list.
 
```
$ node
> c32 = require('c32check')
> c32.c32addressDecode('SP3HDY11ZDHYTMP6A850RW4EM7YQCPFP9WMEXFZHK')
[ 22, 'e2df043f6c7daa58ca41418e11d43faecb3ec9e5' ]
$ curl -s http://52.0.54.100:20443/v2/neighbors | jq | grep -A 10 -b 10 e2df043f6c7daa58ca41418e11d43faecb3ec9e5
      "peer_version": 402653189,
      "port": 41504,
      "public_key_hash": "dfe72d449196474fbd1c7085a4f9887d89ad4599"
    },
    {
      "authenticated": true,
      "ip": "18.208.100.122",
      "network_id": 1,
      "peer_version": 402653189,
      "port": 50384,
      "public_key_hash": "e2df043f6c7daa58ca41418e11d43faecb3ec9e5"
    },
```

**Reply by wileyj** [reply]
for future readers of this discussion - the above works, but you may have to recursively walk many neighbor lists to find the public key you're looking for.
